zstd/blockdec.go

```diff
@@ -123,12 +123,10 @@ func newBlockDec(lowMem bool) *blockDec {
 // Input must be a start of a block and will be at the end of the block when returned.
 func (b *blockDec) reset(br byteBuffer, windowSize uint64) error {
 	b.WindowSize = windowSize
-	tmp := br.readSmall(3)
-	if tmp == nil {
-		if debug {
-			println("Reading block header:", io.ErrUnexpectedEOF)
-		}
-		return io.ErrUnexpectedEOF
+	tmp, err := br.readSmall(3)
+	if err != nil {
+		println("Reading block header:", err)
```

Comment:
Are these printlns meant to be gated with `debug`?

Reply:
@twmb There is another check in the [println](https://github.com/klauspost/compress/blob/a5aad11bec853fa7441dfa47cf5b3e05b7ede0e4/zstd/zstd.go#L91) function, so it will not print unless compiled with debug enabled. 

After encoding and decoding was separated it should, though.

Reply:
Ah thank you very much! I didn't see that this was an override of the Go builtin, but seeing two arguments should've been the hint.